src/main/java/online/partyrun/partyrunmatchingservice/domain/waiting/queue/InMemoryWaitingQueue.java

```diff
@@ -0,0 +1,54 @@
+package online.partyrun.partyrunmatchingservice.domain.waiting.queue;
+
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+
+import online.partyrun.partyrunmatchingservice.domain.waiting.exception.DuplicateUserException;
+import online.partyrun.partyrunmatchingservice.domain.waiting.exception.NotSatisfyCountException;
+import online.partyrun.partyrunmatchingservice.domain.waiting.root.RunningDistance;
+import online.partyrun.partyrunmatchingservice.domain.waiting.root.WaitingUser;
+
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.IntStream;
+
+@Repository
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class InMemoryWaitingQueue implements WaitingQueue {
+    private static final int SATISFY_COUNT = 2;
+    Map<RunningDistance, Queue<String>> map = new ConcurrentHashMap<>();
+
+    public InMemoryWaitingQueue() {
+        Arrays.stream(RunningDistance.values())
+                .forEach(distance -> map.put(distance, new LinkedList<>()));
+    }
+
+    @Override
+    public void add(final WaitingUser user) {
+        if (hasElement(user.userId())) {
+            throw new DuplicateUserException();
+        }
+        map.get(user.distance()).add(user.userId());
+    }
+
+    private boolean hasElement(final String element) {
+        return Arrays.stream(RunningDistance.values())
+                .map(map::get)
+                .anyMatch(q -> q.contains(element));
+    }
```

Comment:
```suggestion
    private boolean hasElement(final String element) {
        return map.values().stream()
                .anyMatch(q -> q.contains(element));
    }
```

인메모리라 어차피 사라질것같긴하지만 몬가 비효율적인거같아서?

Reply:
반영했습니다